cpp/bench/ann/src/raft/raft_ann_bench_utils.h

```diff
@@ -65,16 +63,10 @@ class configured_raft_resources {
    */
   explicit configured_raft_resources(const std::shared_ptr<device_mr_t>& mr)
     : mr_{mr},
-      sync_{[]() {
-              auto* ev = new cudaEvent_t;
-              RAFT_CUDA_TRY(cudaEventCreate(ev, cudaEventDisableTiming));
-              return ev;
-            }(),
-            [](cudaEvent_t* ev) {
-              RAFT_CUDA_TRY_NO_THROW(cudaEventDestroy(*ev));
-              delete ev;
-            }},
-      res_{cudaStreamPerThread}
+      workspace_dev_mr_{},
+      workspace_mr_(&workspace_dev_mr_, 128 * 1024 * 1024ull),
```

Comment:
I don't think we'll be able catch this in our benchmarks. Note, this argument sets the initial size; the maximum stays at its default (which, I think, is half of the total device memory). If 128 MiB is not enough, the workspace will be resized in the first iteration, which will either be omitted as a warm-up iteration or averaged out in the main loop.
I reduced value compared to 1 GiB  of the main pool memory resource, because this caused problems with the large number of threads in the throughput mode, i.e. with 32 threads (33GiB for all workspaces) it simply crashed on my workstation GPU.